### DISCUSSION TITLE
Microweber wrapped around your Laravel app

### DISCUSSION THREAD

**Discussion by ezstoritve** [microweber/microweber#985] [Show and tell]
If someone needs his Laravel App to run inside Microweber with all MW functionalities (modules, admin, CMS, liveedit, ...) i know how to do it. I have integrated relatively big custom app to run together with MW and with blade directives for frontend.

**Comment by gokkstudio** [comment]
I'm interested, do you have a documentation?

**Comment by ezstoritve** [comment]
Hi, i don't have it yet. I am still re-designing some things in MW code so it will be possible to use most of the MW admin functions in my own Laravel project.

But for quick "directions", i have made custom App package in src folder and my own View trait and FrontendController based on MW's one, which is in MW App controllers. I have loaded this App before MW Event and Route provider so i can intercept all routes before MW does. 

'providers' =>
        array (
            0 => 'MicroweberPackages\\App\\Providers\\AppServiceProvider',
            1 => 'EZPackages\\App\\Providers\\AppServiceProvider',
            2 => 'MicroweberPackages\\App\\Providers\\EventServiceProvider',
            3 => 'MicroweberPackages\\App\\Providers\\RouteServiceProvider',
        ),

I have also made some modifications to few MW files to intercept how content is managed (edited and shown) as MW everything loads as modules based on url - first segment is module.

Edited MW files:
MW/App/Providers/AppServiceProvider.php - route loading override to load them in my package as i want to
MW/App/routes/web.php - omit loading MW routes if my package is loaded. General routes for content are loaded/intercepted by my app's frontend controller
MW/Content/ContentManagerHelpers.php - get_by_url to get content by full url not by module
MW/Content/ContentManagerCrud.php - save_from_live_edit to save content by full url not by module
